src/support_enumeration.jl

```diff
@@ -0,0 +1,252 @@
+        A[end, end] = zero(T)
+        b = zeros(T, k+1)
+        b[end] = one(T)
+        while supps[1][end] < nums_actions[1]+1
+            supps[2][:] = collect(1:k)
+            while supps[2][end] < nums_actions[2]+1
+                if _indiff_mixed_action!(A, actions[2], b,
+                                         payoff_matrix1, supps[1], supps[2])
+                    if _indiff_mixed_action!(A, actions[1], b, payoff_matrix2,
+                                             supps[2], supps[1])
+                        out = (zeros(nums_actions[1]),
+                               zeros(nums_actions[2]))
+                        for (p, (supp, action)) in enumerate(zip(supps,
+                                                                 actions))
+                            out[p][supp] = action
+                        end
+                        produce(out)
+                    end
+                end
+                _next_k_array!(supps[2])
+            end
+            _next_k_array!(supps[1])
+        end
+    end
+
+end
+
+"""
+    _indiff_mixed_action!{T<:Real}(A::Matrix{T}, out::Vector{Float64},
+                                   b::Vector{T},
+                                   payoff_matrix::Matrix{T},
+                                   own_supp::Vector{Int},
+                                   opp_supp::Vector{Int})
+
+Given a player's payoff matrix `payoff_matrix`, an array `own_supp`
+of this player's actions, and an array `opp_supp` of the opponent's
+actions, each of length k, compute the opponent's mixed action whose
+support equals `opp_supp` and for which the player is indifferent
+among the actions in `own_supp`, if any such exists. Return `true`
+if such a mixed action exists and actions in `own_supp` are indeed
+best responses to it, in which case the outcome is stored in `out`;
+`false` otherwise. Arrays `A` and `b` are used in intermediate
+steps.
+
+# Arguments
+* `A::Matrix{T}`: Matrix used in intermediate steps.
+* `out::Vector{Float64}`: Vector to store the nonzero values of the 
+    desired mixed action.
+* `b::Vector{T}`: Vector used in intermediate steps.
+* `payoff_matrix::Matrix{T}`: The player's payoff matrix.
+* `own_supp::Vector{Int}`: Vector containing the player's action indices.
+* `opp_supp::Vector{Int}`: Vector containing the opponent's action indices.
+
+# Returns
+* `::Bool`: `true` if a desired mixed action exists and `false` otherwise.
+"""
+function _indiff_mixed_action!{T<:Real}(A::Matrix{T}, out::Vector{Float64},
+                                        b::Vector{T},
+                                        payoff_matrix::Matrix{T},
+                                        own_supp::Vector{Int},
+                                        opp_supp::Vector{Int})
+
+    m = size(payoff_matrix, 1)
+    k = length(own_supp)
+
+    A[1:end-1, 1:end-1] = payoff_matrix[own_supp, :][:, opp_supp]
+    sol = Vector{Float64}(k+1)
+    try
+        sol[:] = A \ b
```

Comment:
Yes, I am thinking of allocating `sol` just once in the loop of `_support_enumeration_task` and passing it to `_indiff_mixed_action!`, similarly to `A` and `b`.